packages/ui/src/hooks/useCredentialChecker.js

```diff
@@ -15,69 +14,65 @@ export const useCredentialChecker = () => {
      * Check if a flow has missing credentials and show modal if needed
      * @param {string|object} flowData - Flow data to check
      * @param {function} onAssign - Callback when credentials are assigned
+     * @param {boolean} forceShow - Force show modal regardless of missing credentials
      * @returns {boolean} Whether modal was shown
      */
-    const checkCredentials = useCallback((flowData, onAssign) => {
-        console.log('🔍 checkCredentials called with:', {
-            flowDataType: typeof flowData,
-            hasOnAssign: !!onAssign,
-            currentModalState: showCredentialModal
-        })
-
+    const checkCredentials = useCallback((flowData, onAssign, forceShow = false) => {
         try {
-            const result = extractMissingCredentials(flowData)
-            console.log('🔍 Credential extraction result:', {
-                hasCredentials: result.hasCredentials,
-                missingCount: result.missingCredentials.length,
-                missingTypes: result.missingCredentials.map((c) => c.credentialType)
-            })
+            let result
+            let credentialsToShow
+
+            if (forceShow) {
+                // For QuickSetup mode, extract ALL credentials (assigned and unassigned)
+                result = extractAllCredentials(flowData)
+                credentialsToShow = result.allCredentials || []
+            } else {
+                // Normal mode - extract only missing credentials
+                result = extractMissingCredentials(flowData)
+                credentialsToShow = result.missingCredentials || []
+            }
 
-            if (result.hasCredentials && result.missingCredentials.length > 0) {
-                console.log('🚨 Missing credentials detected, showing modal')
-                setMissingCredentials(result.missingCredentials)
+            // Show modal if forceShow is true OR if there are missing credentials
+            if (forceShow || (result.hasCredentials && credentialsToShow.length > 0)) {
+                setMissingCredentials(credentialsToShow)
```

Comment:
gets replaced by the `needsSetup` property on the sidekick . This way its done in the server (can be cached) and less data sent to the client.

Reply:
This should also consider that teh backend will need to send information back to the client on the credentialsToShow, so may need more than just a boolean property.